patches/net/minecraft/network/codec/ByteBufCodecs.java.patch

```diff
@@ -13,3 +13,34 @@
              }
  
              public R decode(RegistryFriendlyByteBuf p_331253_) {
+@@ -519,9 +_,17 @@
+         return new StreamCodec<RegistryFriendlyByteBuf, HolderSet<T>>() {
+             private static final int NAMED_SET = -1;
+             private final StreamCodec<RegistryFriendlyByteBuf, Holder<T>> holderCodec = ByteBufCodecs.holderRegistry(p_332137_);
+-
++            private final Map<net.neoforged.neoforge.registries.holdersets.HolderSetType, StreamCodec<RegistryFriendlyByteBuf, net.neoforged.neoforge.registries.holdersets.ICustomHolderSet<T>>> holderSetCodecs = new java.util.HashMap<>();
++
++            private StreamCodec<RegistryFriendlyByteBuf, net.neoforged.neoforge.registries.holdersets.ICustomHolderSet<T>> holderSetCodec(net.neoforged.neoforge.registries.holdersets.HolderSetType type) {
```

Comment:
Feels like this really ought to be a `StreamCodec<RegistryFriendlyByteBuf, ? extends ICustomHolderSet<T>>` instead -- then you don't need the `cast` ugliness.